File: typespec/employer/profilepage.ts
export interface GetHubUserBioRequest {
  handle: string;
}

export interface EmployerWorkHistory {
  id: string;
  employer_domain: string;
  employer_name?: string;
  title: string;
  start_date: Date;
  end_date?: Date;
  description?: string;
}

export interface EmployerViewBio {
  handle: string;
  full_name: string;
  short_bio: string;
  long_bio: string;
  verified_mail_domains?: string[];
  work_history: EmployerWorkHistory[];
}

File: typespec/employer/profilepage.go
package employer

type GetHubUserBioRequest struct {
	Handle string `json:"handle" validate:"required"`
}

type EmployerWorkHistory struct {
	ID             string  `json:"id"`
	EmployerDomain string  `json:"employer_domain"`
	EmployerName   *string `json:"employer_name"`
	Title          string  `json:"title"`
	StartDate      string  `json:"start_date"`
	EndDate        *string `json:"end_date"`
	Description    *string `json:"description"`
}

type EmployerViewBio struct {
	Handle              string                `json:"handle"`
	FullName            string                `json:"full_name"`
	ShortBio            string                `json:"short_bio"`
	LongBio             string                `json:"long_bio"`
	VerifiedMailDomains []string              `json:"verified_mail_domains,omitempty"`
	WorkHistory         []EmployerWorkHistory `json:"work_history"`
}
